### PR TITLE
[ISSUE #4583] Fix DOMAIN mode retry logic

### PR DESCRIPTION
**Pull request by lexburner**
## What is the purpose of the change

fix https://github.com/alibaba/nacos/issues/4583

## Brief changelog

fix namingProxy domain retry logic; 
add retry param namingRequestDomainCount

### CODE REVIEW

**Review by KomachiSion — APPROVED** [review]

**Comment by KomachiSion** [comment]
Please fix commit user problem and sign the CLA.

**Comment by lexburner** [comment]
> Please fix commit user problem and sign the CLA.

solved

File: client/src/main/java/com/alibaba/nacos/client/naming/net/NamingProxy.java [hunk 1]
```diff
@@ -513,7 +519,20 @@ public String reqApi(String api, Map<String, String> params, Map<String, String>
         }
         
         NacosException exception = new NacosException();
-        
+
+        if (StringUtils.isNotBlank(nacosDomain)) {
+            for (int i = 0; i < UtilAndComs.REQUEST_DOMAIN_RETRY_COUNT; i++) {
```
Comment (horizonzy) [inline client/src/main/java/com/alibaba/nacos/client/naming/net/NamingProxy.java:524]: Where is the place `maxRetry ` used?  The count limit should be `maxRetry`

Comment (lexburner) [reply client/src/main/java/com/alibaba/nacos/client/naming/net/NamingProxy.java]: my fault, fixed.

**Review by horizonzy — APPROVED** [review]
LGTM

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**